### PR TITLE
Fleece API update

### PR DESCRIPTION
**Pull request by jayahariv**
* Replace the FLValue_FromData() with FLDoc_FromData()
* Include FLExpert.hh for using the AllocedDict

### CODE REVIEW

File: Objective-C/Internal/CBLFleece.hh [hunk 1]
```diff
@@ -51,7 +51,7 @@ namespace cbl {
     double    asDouble  (const fleece::MValue<id>&, const fleece::MCollection<id> &container);
     
     // parses the JSON string, into FLValue
-    FLValue parseJSON(const FLSlice json, NSError** error);
+    id parseJSON(const FLSlice json, NSError** error);
```
Comment (pasin) [inline Objective-C/Internal/CBLFleece.hh:54]: Comment should be changed as well.

**Review by pasin — COMMENTED** [review]
Build is still failed.

**Comment by jayahariv** [comment]
For the CI job failure, its already fixed in a different PR

File: Objective-C/Internal/CBLFleece.hh [hunk 2]
```diff
@@ -51,7 +51,7 @@ namespace cbl {
     double    asDouble  (const fleece::MValue<id>&, const fleece::MCollection<id> &container);
     
     // parses the JSON string, into FLValue
```
Comment (jayahariv) [inline Objective-C/Internal/CBLFleece.hh:53]:
```suggestion
    // parses the JSON string, into NSObject(NSArray, NSDictionary)
```

**Review by pasin — APPROVED** [review]

**Pull request merged**